Gordon360/Services/RecIM/ActivityService.cs

```diff
@@ -0,0 +1,146 @@
+﻿using Gordon360.Models.CCT;
+using Gordon360.Models.ViewModels.RecIM;
+using Gordon360.Models.CCT.Context;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+
+namespace Gordon360.Services.RecIM
+{
+    public class ActivityService : IActivityService
+    {
+        private readonly CCTContext _context;
+        private readonly ISeriesService _seriesService;
+
+        public ActivityService(CCTContext context, ISeriesService seriesService)
+        { 
+            _context = context;
+            _seriesService = seriesService;
+        }
+
+
+        public IEnumerable<ActivityViewModel> GetActivities()
+        {
+            var activities = _context.Activity
+                            .Select(a => new ActivityViewModel
+                            {
+                                ID = a.ID,
+                                Name = a.Name,
+                                RegistrationStart = a.RegistrationStart,
+                                RegistrationEnd = a.RegistrationEnd,
+                                Sport = _context.Sport
+                                        .FirstOrDefault(s => s.ID == a.SportID),
+                                Status = _context.ActivityStatus
+                                        .FirstOrDefault(s => s.ID == a.StatusID)
+                                        .Description,
+                                MinCapacity = a.MinCapacity,
+                                MaxCapacity = a.MaxCapacity,
+                                SoloRegistration = a.SoloRegistration,
+                                Logo = a.Logo,
+                                Completed = a.Completed,
+                                Series = a.Series
+                                        .Select(s => new SeriesViewModel
+                                        {
+                                            ID = s.ID,
+                                            Name = s.Name,
+                                            StartDate = s.StartDate,
+                                            EndDate = s.EndDate,
+                                            Type = _context.SeriesType
+                                                    .FirstOrDefault(st => st.ID == s.TypeID)
+                                                    .Description
+                                        }).ToList()
+                            });
+            return activities;
+        }
+        public IEnumerable<ActivityViewModel> GetActivitiesByTime(DateTime? time)
+        {
+            if (time is null)
+            {
+                return GetActivities().Where(a => !a.Completed);
+            }
+            else
+            {
+                return GetActivities().Where(a => a.RegistrationEnd > time);
+            }
+        }
```

Comment:
This method is very confusing as to what you are trying to return (just by looking at the field names).

Reply:
this is connected to the GetActivities route which has 2 optional parameters:
active
time

if neither are filled, the controller calls GetAllActivities with no parameters,
if active, GetActivitiesByTime is called without a parameter
if time has a value, then we call GetActivitiesByTime with a time parameter

Reply:
I didn't elaborate since I was rushing to get this finished before my 3pm yesterday, but What I'm confused about is how it is filtering by `RegistrationEnd > time`. It feels like there should also be a `RegistrationStart < time` here as well at the least.